### ISSUE TITLE
use SVectors?

### ISSUE THREAD

**Issue by stevengj** [IgorKohan/NormalHermiteSplines.jl#3]
Since a given spline object is for a fixed dimension `n`, you should strongly consider using `SVector{n}` objects from StaticArrays.jl to represent points in ℝⁿ.

That would probably simplify your code — you can use vector operations for some things that are currently explicit loops, and some of your explicit loops seem to be in the [wrong order for locality](https://github.com/IgorKohan/NormalHermiteSplines.jl/blob/63077502eb1f8579cb6789a14bd291355c76a4a5/src/Interpolate.jl#L157), which would be automatically fixed for `SVector` operations.   Some of your [dimension checks](https://github.com/IgorKohan/NormalHermiteSplines.jl/blob/63077502eb1f8579cb6789a14bd291355c76a4a5/src/Interpolate.jl#L188-L190) could be eliminated because they would be enforced by the type system.

It should also remove many of your allocations, e.g. `norm(ξ .- η)` allocates a temporary vector [here](https://github.com/IgorKohan/NormalHermiteSplines.jl/blob/63077502eb1f8579cb6789a14bd291355c76a4a5/src/ReproducingKernels.jl#L92), which is especially concerning since `_rk` seems to be a performance-critical inner-loop function in your spline evaluation.

Note also that slices allocate in Julia unless you use the `@views` macro, so lines like [this inner loop](https://github.com/IgorKohan/NormalHermiteSplines.jl/blob/63077502eb1f8579cb6789a14bd291355c76a4a5/src/Interpolate.jl#L605) are allocating.  However, if e.g. `points` were an array of `SVector{n}`s, you could do `points[p]` instead of `points[:,p]` and avoid the heap allocation.

Basically, this seems like a perfect use-case for SVectors.

**Comment by IgorKohan** [comment]
Thank you Steven for your suggestion and code review! I'll fix these issues in the next release.
Regards,
Igor
